packages/app/src/cli/models/extensions/specification.ts

```diff
@@ -104,22 +117,35 @@ export class UIExtensionInstance<TConfiguration extends BaseConfigContents = Bas
     return this.specification.isPreviewable
   }
 
+  get features(): ExtensionFeature[] {
+    return this.specification.appModuleFeatures(this.configuration)
+  }
+
+  set usingExtensionsFramework(value: boolean) {
+    this._usingExtensionsFramework = value
+  }
+
   constructor(options: {
     configuration: TConfiguration
     configurationPath: string
-    entryPath: string
+    entryPath?: string
     directory: string
-    specification: UIExtensionSpec
+    specification: ExtensionSpecification
   }) {
     this.configuration = options.configuration
     this.configurationPath = options.configurationPath
-    this.entrySourceFilePath = options.entryPath
+    this.entrySourceFilePath = options.entryPath ?? ''
     this.directory = options.directory
     this.specification = options.specification
-    this.outputBundlePath = joinPath(options.directory, 'dist/main.js')
     this.devUUID = `dev-${randomUUID()}`
     this.localIdentifier = basename(options.directory)
     this.idEnvironmentVariableName = `SHOPIFY_${constantize(basename(this.directory))}_ID`
+    this._usingExtensionsFramework = false
+    if (this.specification.identifier === 'theme') {
+      this.outputBundlePath = this.directory
+    } else {
+      this.outputBundlePath = joinPath(this.directory, 'dist/main.js')
+    }
```

Comment:
maybe we could still have helper functions that run this logic for you? I feel that now this file has ALL the logic from different extension types

Reply:
Yes, there is some logic that I added here that's specific to some extensions, I plan to move this out and simplify stuff. This case is interesting, because we shouldn't need to set the outputBundlePath here, I'll improve this in future PRs 🙏